tests/shared/src/test/scala/cats/tests/ParallelSuite.scala

```diff
@@ -390,6 +390,38 @@ class ParallelSuite
     }
   }
 
+  test("ParFlatMapN over List should be consistent with parMapN flatten") {
+    forAll { (as: List[Int], bs: List[Int], cs: List[Int]) =>
+      val mf: (Int, Int, Int) => List[Int] = _ :: _ :: _ :: Nil
+      assert((as, bs, cs).parFlatMapN(mf) == (as, bs, cs).parMapN(mf).flatten)
+    }
+  }
+
+  test("ParFlatMapN over NonEmptyList should be consistent with parMapN flatten") {
+    forAll { (as: NonEmptyList[Int], bs: NonEmptyList[Int], cs: NonEmptyList[Int]) =>
+      val mf: (Int, Int, Int) => NonEmptyList[Int] = (a, b, c) => NonEmptyList.of(a, b, c)
+      assert((as, bs, cs).parFlatMapN(mf) == (as, bs, cs).parMapN(mf).flatten)
+    }
+  }
+
+  test("ParFlatMap over List should be consistent with flatmap") {
+    forAll { as: List[Int] => assert(Tuple1(as).parFlatMap(_ :: Nil) == Tuple1(as).flatMap(_ :: Nil)) }
+  }
+
+  test("ParFlatMap over NonEmptyList should be consistent with flatmap") {
+    forAll { as: NonEmptyList[Int] =>
+      assert(Tuple1(as).parFlatMap(NonEmptyList.one) == Tuple1(as).flatMap(NonEmptyList.one))
+    }
+  }
+
+  test("ParMapN over f should be consistent with parFlatMapN over f lifted in Kleisli") {
+    forAll { (as: List[Int], bs: List[Int]) =>
+      val f: (Int, Int) => Int = _ + _
+      val mf: (Int, Int) => List[Int] = Function.untupled(Kleisli.fromFunction[List, (Int, Int)](f.tupled).run)
```

Comment:
I'm not sure what Kleisli is doing here. Couldn't we just do: `val mf: (Int, Int) => List[Int] = { (a, b) => List(a + b) }`?

Reply:
not very much, my intention was to show that `_.parMapN(f)` is equivalent to `_.parFlatMapN(mf)` where `mf` is `f: A => B` lifted in a `Kleisli[M, A, B]` where `Parallel[M]` exists. It turned out bad since I had to tuple/untuple unluckily (`Kleisli.fromFunction(f).run` is way more readable). For the sake of readability I'll use a cleaner impl of that function, gimme sec :D

Reply:
Just pushed this
```scala
val f: (Int, Int) => Int = (a, b) => a + b
val mf: (Int, Int) => List[Int] = (a, b) => List(a + b)
```